libraries/core-react/src/components/TableOfContents/TableOfContents.stories.tsx

```diff
@@ -179,3 +187,153 @@ export const Example: Story<TableOfContentsProps> = (args) => {
     </Wrapper>
   )
 }
+
+export const Compact: Story<TableOfContentsProps> = (args) => {
+  const [density] = useState<Density>('comfortable')
+
+  return (
+    <EdsProvider density={density}>
+      <Wrapper>
+        <main>
+          <article>
+            <Typography variant="h1">Heading 1</Typography>
+            <Typography variant="body_long">
+              Lorem ipsum dolor sit amet consecteur dit lot. Lorem ipsum dolor
+              sit amet consecteur dit lot. Lorem ipsum dolor sit amet consecteur
+              dit lot.
+            </Typography>
+            <Typography variant="body_long">
+              Lorem ipsum dolor sit amet consecteur dit lot. Lorem ipsum dolor
+              sit amet consecteur dit lot. Lorem ipsum dolor sit amet consecteur
+              dit lot.
+            </Typography>
+            <Typography variant="body_long">
+              Lorem ipsum dolor sit amet consecteur dit lot. Lorem ipsum dolor
+              sit amet consecteur dit lot. Lorem ipsum dolor sit amet consecteur
+              dit lot.
+            </Typography>
+            <Typography variant="h2" id="sub-section-one">
```

Comment:
These id's need to change so they are not duplicated from the other story, otherwise it breaks the functionality